### ISSUE TITLE
Garbage output with pdf mode and not txt

### ISSUE THREAD

**Issue by Sloth6** [tesseract-ocr/tesseract#1238]
### Environment

* **Tesseract Version**: 
tesseract 4.00.00alpha
 leptonica-1.74.4
  libjpeg 9b : libpng 1.6.34 : libtiff 4.0.9 : zlib 1.2.8
 Found AVX
 Found SSE

* **Platform**: 
Darwin MacBook-Pro.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64

[Also replicated on Ubuntu 14.04 with tesseract 4.00.00alpha]

### Current Behavior:
The text output is garbage when using pdf mode and perfect when using txt mode.

commands run...

> $ tesseract cropped.tiff cropped txt
$ tesseract cropped.tiff cropped pdf
$ gs -sDEVICE=txtwrite -o cropped_dump.txt cropped.pdf

[debug_files.zip](https://github.com/tesseract-ocr/tesseract/files/1573511/debug_files.zip)

This is what the output from the txt output looks like. 

> ABBREVIATIONS
ABBREVIATIONS
ABBREVIATIONS
ACCESSIBILITY CLEARANCES
ACCESSIBILITY CLEARANCES
STANDARD SYMBOLS
...

and from a ghostscript dump out the output pdf 

>                              
                      SNOILVIATYE@V
                                                  
          ©        IY
           LOY       INLOY
           ‘OI1V     AO9Y
           ‘44V      3AO8Y   100714
          VIV          NYOMN3WY3LNLLLSNI140
           331Y        NYOMNAWY10   T¥9IML9973
           oSIVY       NYOIM3WY3LNLILSNI140713418NOILONMLSNOD
          TV           WNNIWATY
           ‘NIV      AXYVIY
           1Y          41YNMTLTY
           dWy
           ‘LdYf        LNJWILNVIY
           ‘HONY
            IVNHSV     NYOIM3WYAL3190810‘SNILYIH‘ONILYX39IM43Y
                  8        ONINOILIQNOS

It is not an issue with ghostscript pdf to text since copy-pasting text from pdf is the same. The TIFF is 300ppi and each character is about 30px high.

### Expected Behavior:
Text layer of pdf being somewhat similar to txt output or at least partially correct. 

### Suggested Fix:
None

Any help is greatly appreciated.

**Comment by jbreiden** [comment]
Text extraction on Chrome also shows similar garbage. Using `qpdf --stream-data=uncompress` the first few lines of embedded text looks like this. Thus the first three characters should be ascii 4E 4E 49 do not look very promising to me ('O' 'N' 'I'). Not sure what is going on.

```
BT
3 Tr -1 0 0 -1 214.56 7.2 Tm /f-0-0 10 Tf 93.2 Tz [ <004F004E0049004E004F0049004C00490051004E004F0053> ] TJ 73.44 0 Td 91.2 Tz [ <0038> ] TJ 
 -189.12 -11.28 Td 91.2 Tz [ <2018004F004E0049004C00590058003300390049004D003400330059> ] TJ 67.44 0 Td 87 Tz [ <20180053004E0049004C005900490048> ] TJ 37.68 0 Td 100.8 Tz [ <00310030> ] TJ 12.72 0 Td 94.628 Tz [ <0041004C00330031003900300038> ] TJ 36.24 0 Td 100.2 Tz [ <004E0059004F0049004D003300570059> ] TJ 80.88 0 Td 104 Tz [ <00490056004E004800530056> ] TJ 
```

**Comment by jbreiden** [comment]
Problem reproduces with the copy of Tesseract I have in hand, which is not currently synchronized with git. So whatever is going on, it is not brand new.

**Comment by jbreiden** [comment]
Found it. The configuration for PDF creation contains an additional directive, which sets page segmentation mode to 1. When this is set, Tesseract gets confused by this image and tries to recognize it as upside down. Removing that directive makes text and pdf recognition match. That is your workaround. I do not recall what page_seg_mode 1 does nor why we have it default to on for PDF and off for text.

`config/txt` 
```tessedit_create_txt 1```

`config/data` 
```
tessedit_create_pdf 1
tessedit_pageseg_mode 1
```

**Comment by jbreiden** [comment]
Here's the relevant help for page segmentation mode.

  1    Automatic page segmentation with OSD.
  3    Fully automatic page segmentation, but no OSD. (Default)

**Comment by jbreiden** [comment]
I'm going to mark fixed because this is understood. I consider it an open question is we want to make any changes to configs, but probably not bug-worthy.

**Comment by Sloth6** [comment]
Hi jbreiden, 

Thank you!! An equally fast and helpful reply. That fixed the problem completely. For anyone else who finds this, you need to pass this config file as an option...

>  tessedit_create_pdf 1
tessedit_pageseg_mode 3

**Comment by stweil** [comment]
Fixed with PR #1943.
